File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/typescript/F11.P1.checked.ts
function readAtChecked(xs: number[], i: number): number {
// BEGIN PROBE F11.P1.checked
const e = xs.at(i) ?? 0
return e
// END PROBE F11.P1.checked
}

console.log(readAtChecked([1, 2, 3], 2))

File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/rust/F11.P1.checked.rs
fn read_at(xs: Vec<i32>, i: i32) -> i32 {
// BEGIN PROBE F11.P1.checked
let e = *xs.get(i as usize).unwrap_or(&0);
e
// END PROBE F11.P1.checked
}

fn main() {
    println!("{}", read_at(vec![1, 2, 3], 2));
}
